lib/Basic/Platform.cpp

```diff
@@ -118,6 +118,12 @@ StringRef swift::getPlatformNameForTriple(const llvm::Triple &triple) {
   case llvm::Triple::FreeBSD:
     return "freebsd";
   case llvm::Triple::Win32:
+    if (triple.isWindowsCygwinEnvironment())
+      return "cygwin";
+
+    if (triple.isWindowsGNUEnvironment())
+      return "mingw";
+
```

Comment:
Did you mean a nested switch in `case llvm::Triple::Win32:` ?
```
switch (triple.getEnvironment()) {
case llvm::Triple::Cygnus:
  return "cygwin";
case llvm::Triple::GNU:
  return "mingw":
default:
  return "windows";
}
```
I can not imagine how cover those with single switch.